src/main/java/org/folio/innreach/domain/service/impl/ContributionActionServiceImpl.java

```diff
@@ -95,45 +101,58 @@ public void handleItemCreation(Item newItem) {
+    log.info("Handling item creation {}", newItem.getId());
+
+    var instance = fetchInstanceWithItems(newItem);
+    if (!isMARCRecord(instance)) {
+      ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, MARC_ERROR_MSG, FAILED);
+      return;
+    }
+    var centralServerId = ongoingContributionStatus.getCentralServerId();
+    if(checkCentralServerValid(centralServerId)) {
+      contributionJobRunner.runItemContribution(centralServerId, instance, newItem, ongoingContributionStatus);
+    } else {
+      ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, INVALID_CENTRAL_SERVER_ID, FAILED);
+    }
+  }
 
   @Override
-  public void handleItemUpdate(Item newItem, Item oldItem) {
+  public void handleItemUpdate(Item newItem, Item oldItem, OngoingContributionStatus ongoingContributionStatus) {
     log.info("Handling item update {}", newItem.getId());
 
     var instance = fetchInstanceWithItems(newItem);
     if (!isMARCRecord(instance)) {
+      ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, MARC_ERROR_MSG, FAILED);
       return;
     }
 
     var oldInstance = fetchOldInstance(newItem, oldItem, instance);
     boolean itemMoved = !oldInstance.getId().equals(instance.getId());
-
-    handlePerCentralServer(newItem.getId(), csId -> {
+    var centralServerId = ongoingContributionStatus.getCentralServerId();
+    if(checkCentralServerValid(centralServerId)) {
       if (itemMoved) {
-        contributionJobRunner.runItemMove(csId, instance, oldInstance, newItem);
+        contributionJobRunner.runItemMove(centralServerId, instance, oldInstance, newItem, ongoingContributionStatus);
       } else {
-        contributionJobRunner.runItemContribution(csId, instance, newItem);
+        contributionJobRunner.runItemContribution(centralServerId, instance, newItem, ongoingContributionStatus);
       }
-    });
+    } else {
+      ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, INVALID_CENTRAL_SERVER_ID, FAILED);
+    }
   }
 
 
   @Override
-  public void handleItemDelete(Item deletedItem) {
+  public void handleItemDelete(Item deletedItem, OngoingContributionStatus ongoingContributionStatus) {
     log.info("Handling item delete {}", deletedItem.getId());
 
     var instance = fetchInstanceWithItems(deletedItem);
     if (!isMARCRecord(instance)) {
+      ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, MARC_ERROR_MSG, FAILED);
       return;
     }
-    try {
-      for (var csId : getCentralServerIds()) {
-        contributionJobRunner.runItemDeContribution(csId, instance, deletedItem);
-      }
-    } catch (ServiceSuspendedException | FeignException | InnReachConnectionException e) {
-      log.info("exception thrown from handleItemDelete", e);
-      throw e;
-    }
```

Comment:
I think moving these to the highest level where they could occur is the right choice.

Reply:
Actually, we are removing this logic because whenever we are persisting ongoing event, we are persisting the event against every central serverId. So this check is not necessary in this place. hence, the code is removed.